app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb

```diff
@@ -61,14 +61,24 @@ def allowed_customization_templates(options = {})
     end
   end
 
-  def allowed_hosts_obj(options = {})
-    return [] if (src = resources_for_ui).blank?
-    return [] unless src[:vm]
+  def allowed_datacenters(_options = {})
+    datacenter_by_vm ? super.slice(datacenter_by_vm.id) : super
```

Comment:
If `datacenter_by_vm` is nil super here will return an empty hash.
You could also do this with which reads cleaner:`super.slice(datacenter_by_vm.try(:id))`